tools/x-docs/src/docs/guides/components/cookbook/styling.md

```diff
@@ -1,7 +1,94 @@
+Some x-dash components, such as [x-teaser](/components/x-teaser/readme.md), are built to be compatible with styles provided by Origami components, by using their classnames. These styles ase expected to be provided by the application that's consuming the x-dash component, for example by including the CSS from the [Origami Build Service](https://www.ft.com/__origami/service/build/v2/). To include these styles in a component's demos in the documentation and component explorer, use the `dependencies` object in your component's [`stories/index.js`](/tools/x-docs/guides/components/setup.md#indexjs).
 
-We're actively investigating better-integrated ways of including styling with components.
+## Component-specific styles
+
+Some components may require styling not provided by existing Origami components. If your component's styles might be useful outside of the FT.com ecosystem, you may want to speak to the [Origami team](http://origami.ft.com/) about creating its styles as an Origami component.
+
+If your component will only be used within FT.com, you can include its styles as CSS Modules within the component.
+
+### CSS Modules
+
+A CSS Module is a self-contained CSS file that can be used with ECMAScript `import`. When a component is bundled, its CSS Modules are bundled into a single `.css` file, and their classes are obfuscated into effectively-random strings, so styles from a component cannot interfere with any other part of a page they're included into. The `default` export of a CSS Module is an object containing its classes as keys, and their obfuscated versions as values. This allows you to reference the classes from your component by their readable names, and output the obfuscated names when the component is used.
+
+Keep your CSS files in your `src` directory, adjacent to the component that will use them, with the same naming convention as your `.jsx` files. For example, if you have a `Button` component in `Button.jsx`, its styles should be in the same directory, in a file called `Button.css`.
+
+```css
+/* Button.css */
+
+.button {
+	background: steelblue;
+	color: white;
+	border-radius: 0.25em;
+	padding: 0.5em 1em;
+}
+
+.large {
+	font-size: 1.6em;
+}
+
+.danger {
+	background: firebrick;
+}
+```
+
+Although these classes are short and generic, there is no risk of them interfering with anything else on the page, because they'll be obfuscated into strings like `.Button_button__vS7Mv`.
+
+When you `import` this CSS file, you can reference its styles using the object it exports:
+
+```jsx
+// Button.jsx
+
+import { h } from '@financial-times/x-engine';
+import buttonStyles from './Button.css';
+
+export const Button = ({large, danger}) => <button
+	className={
+		`${buttonStyles.button} ${large ? buttonStyles.large : ''}  ${danger ? buttonStyles.danger : ''}`
+	}
+>Click me!</button>;
```

Comment:
This example could be made more readable/approachable with different formatting, maybe a bit too contrived? Good explanation below - perhaps worth adding to the example code as well?

Reply:
got any suggestions for a better example?

Reply:
How about:

```jsx
export const Button = ({large, danger}) => {
	const classNames = [
		buttonStyles.button,
		large ? buttonStyles.large : '',
		danger ? buttonStyles.danger : ''
	];

	return <button className={classNames.join(' ')}>Click me!</button>;
};
```